src/main/java/frc/robot/hardware/motor/phoenix6/TalonFXMotor.java

```diff
@@ -0,0 +1,67 @@
+package frc.robot.hardware.motor.phoenix6;
+
+import com.ctre.phoenix6.signals.NeutralModeValue;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.wpilibj2.command.sysid.SysIdRoutine;
+import frc.robot.hardware.motor.ControllableMotor;
+import frc.robot.hardware.phoenix6.Phoenix6Device;
+import frc.robot.hardware.request.IRequest;
+import frc.robot.hardware.request.phoenix6.Phoenix6AngleRequest;
+import frc.robot.hardware.request.phoenix6.Phoenix6DoubleRequest;
+import frc.utils.calibration.sysid.SysIdCalibrator;
+
+public class TalonFXMotor extends Phoenix6Device implements ControllableMotor {
+
+	private final TalonFXWrapper motor;
+	private final SysIdCalibrator.SysIdConfigInfo sysidConfigInfo;
+
+	public TalonFXMotor(String logPath, TalonFXWrapper motor, SysIdRoutine.Config sysidConfig) {
+		super(logPath);
+		this.motor = motor;
+		this.sysidConfigInfo = new SysIdCalibrator.SysIdConfigInfo(sysidConfig, true);
+		motor.optimizeBusUtilization();
+	}
+
+	@Override
+	public SysIdCalibrator.SysIdConfigInfo getSysidConfigInfo() {
+		return sysidConfigInfo;
+	}
+
+	@Override
+	public void setBrake(boolean brake) {
+		NeutralModeValue neutralModeValue = brake ? NeutralModeValue.Brake : NeutralModeValue.Coast;
+		motor.setNeutralMode(neutralModeValue);
+	}
+
+	@Override
+	public void resetPosition(Rotation2d position) {
+		motor.setPosition(position.getRotations());
+	}
+
+
+	@Override
+	public void stop() {
+		motor.stopMotor();
+	}
+
+	@Override
+	public void setPower(double power) {
+		motor.set(power);
+	}
+
+
+	@Override
+	public void applyDoubleRequest(IRequest<Double> request) {
+		if (request instanceof Phoenix6DoubleRequest) {
+			motor.setControl(((Phoenix6DoubleRequest) request).getControlRequest());
+		}
+	}
+
+	@Override
+	public void applyAngleRequest(IRequest<Rotation2d> request) {
+		if (request instanceof Phoenix6AngleRequest) {
+			motor.setControl(((Phoenix6AngleRequest) request).getControlRequest());
+		}
+	}
```

Comment:
in all of the other motors they require different מימוש, only in talonfx no (for now and current phoenix version). so instead of doing if else i prefer to separate this

Reply:
Even if you have different implementation, it could still help, as it would allow you to not have to "Smoch Al Smoch", you can have the "if" with the types within the function and then check for the correct logic, and this will allow you to handle the case of an invalid type in a much better way and to not trush "Smoch Al Smoch".

I will say, that because of the actual use case for us, I doubt it will matter, but I'm talking about it mainly because you hated the "Smoch Al Smoch" (and I agree with you).

Reply:
Pr for the solution